File: problems/2025/09-23/queue-reversal/solution.js
class Solution {
  reverseQueue(q) {
    const st = [];
    while (q.length) st.push(q.shift());
    while (st.length) q.push(st.pop());
  }
}

const sol = new Solution();
const q = [5,10,15,20,25];
sol.reverseQueue(q);
console.log(q); // [25,20,15,10,5]

File: problems/2025/09-23/queue-reversal/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:
    void reverseQueue(queue<int> &q) {
        stack<int> st;
        while (!q.empty()) {
            st.push(q.front());
            q.pop();
        }
        while (!st.empty()) {
            q.push(st.top());
            st.pop();
        }
    }
};

int main() {
    queue<int> q;
    for (int x : {5,10,15,20,25}) q.push(x);
    Solution().reverseQueue(q);
    while (!q.empty()) { cout << q.front() << (q.size()>1?' ':'\n'); q.pop(); }
    return 0;
}
